### DISCUSSION TITLE
Borderless ComboBox

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI#2331] [Ideas]
Would it be possible to add a borderless styleType to the ComboBox? After converting some borderless dropdowns to comboboxes, we noticed this is not an option.

**Comment by mayank99** [comment]
Can you share your use cases? Screenshots would help.

iTwinUi doesn't provide a borderless `ComboBox` because the concept doesn't make a lot of sense. The user needs to be able to see the input so that they can type into it. iTwinUI provides a [borderless `Select`](https://itwinui.bentley.com/docs/select#borderless), but it is to be used sparingly, and really only recommended when you have a default selected value.

**Reply by HaveSpacesuit** [reply]
Our developers were working on implementing this wireframe and used a borderless select

![image](https://github.com/user-attachments/assets/030903f2-321a-49bf-8418-1e03700cbc52)

However, the UX team decided to change it to a combobox after learning there would be too many options in the dropdown to make a normal select useful.

If I understand you correctly, you would say that if we expect users to click in it and type, then we should just go ahead and display the borders all the time?

**Reply by mayank99** [reply]
That's correct. Input fields should not be displayed without borders. If you could ask your UX designer to reach out to us, we can have a conversation with them to better understand their thinking.

There is a slightly different combobox pattern we might want to support in the future, where the input field appears inside the "dropdown"/popover. We don't currently have any plans to work on this (it's merely an idea), but if we do support it in the future, that would allow the main trigger button to remain borderless.
